File: frontend/src/api/workflow.ts
import API from './client';

export interface Approver {
  userId: string;
  name: string;
  email: string;
  role: string;
}

export interface WorkflowLevel {
  level: number;
  approvers: Approver[];
}

export interface WorkflowRequest {
  companyId: string;
  name: string;
  levels: WorkflowLevel[];
}

export interface WorkflowResponse {
  id: string;
  companyId: string;
  name: string;
  levels: WorkflowLevel[];
  createdBy: string;
  createdAt: string;
  updatedAt: string;
}

export const workflowApi = {
  saveWorkflow: async (data: WorkflowRequest) => {
    const response = await API.post('/workflows', data);
    return response.data;
  },
  
  getWorkflowByCompany: async (companyId: string): Promise<WorkflowResponse> => {
    const response = await API.get(`/workflows/${companyId}`);
    return response.data;
  },
};

File: backend/app/schemas/workflow.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class Approver(BaseModel):
    userId: str
    name: str
    email: str
    role: str

class WorkflowLevel(BaseModel):
    level: int
    approvers: List[Approver]

class CreateWorkflowRequest(BaseModel):
    companyId: str
    name: str
    levels: List[WorkflowLevel]

class WorkflowResponse(BaseModel):
    id: str
    companyId: str
    name: str
    levels: List[WorkflowLevel]
    createdBy: str
    createdAt: datetime
    updatedAt: datetime
